### PR TITLE
[APP] Tapp-103: CP initial implementation

### PR DESCRIPTION
**Pull request by michellemtchai**
**Please merge this after PR #105**

Please see Issue #103 for details.

### CODE REVIEW

File: api/db/migrate/20190516205625_add_columns_to_assignment.rb
Comment (chrislessard) [inline api/db/migrate/20190516205625_add_columns_to_assignment.rb:5]: It's unclear to me what is meant by "pay1" and "pay2". Can this be clarified with a comment or change of names ?

File: api/db/migrate/20190516205813_add_columns_to_offer.rb
Comment (chrislessard) [inline api/db/migrate/20190516205813_add_columns_to_offer.rb:5]: Same as above.

File: api/db/schema.rb
Comment (chrislessard) [inline api/db/schema.rb:46]: Same as above.

File: api/db/schema.rb
Comment (chrislessard) [inline api/db/schema.rb:78]: Same as above.

File: api/app/controllers/api/v1/assignments_controller.rb
Comment (chrislessard) [inline api/app/controllers/api/v1/assignments_controller.rb]: Please fix this formatting.

File: api/app/assets/javascripts/api/v1/assignments.coffee
Comment (chrislessard) [inline api/app/assets/javascripts/api/v1/assignments.coffee]: Why are we generating .coffee files ? Are there plans to use this ? I would prefer these to be dropped these if not.

**Review by chrislessard — CHANGES_REQUESTED** [review]
Hi Michelle, 

Thanks for all this great work ! I do have a few other nits : 

1. Please follow the commit format mentioned in the project README. Yours aren't far off, but we try to indicate both the nature of the commit and the affected region in the header, such as in "fix(docker): add PGDATA to .env" to make the commit long more consistent and clear. 

2. This is creeping towards the upper bound for PR size, as it becomes harder to review everything thoroughly in a reasonable time frame. IMO it would have been ok to keep the DB stuff and the route/code/test boilerplate separate, especially since DB work tends to be more contested.

File: api/db/migrate/20190516205625_add_columns_to_assignment.rb
Comment (michellemtchai) [reply api/db/migrate/20190516205625_add_columns_to_assignment.rb:5]: Actually, I plan to remove these attributes. Even though it was discussed during the meeting on Thursday that we should use them, there's apparently a plan to move them to a new table called `fund_source`. I'm kind of waiting for a final decision in case there's anymore changes...

File: api/app/controllers/api/v1/assignments_controller.rb
Comment (michellemtchai) [reply api/app/controllers/api/v1/assignments_controller.rb]: Do you mean formatting as in the tabbing?

File: api/app/assets/javascripts/api/v1/assignments.coffee
Comment (michellemtchai) [reply api/app/assets/javascripts/api/v1/assignments.coffee]: These were generated when I made the models in rails. Do you have a list of files that should be removed if I generate model through the rails command? Or is it preferred that I make these files manually?

**Comment by michellemtchai** [comment]
I've rewritten the commit messages. Does this look okay now?

File: api/app/assets/javascripts/api/v1/assignments.coffee
Comment (michellemtchai) [reply api/app/assets/javascripts/api/v1/assignments.coffee]: I've removed the non-controller files. Does it look okay now?

File: api/db/migrate/20190516205625_add_columns_to_assignment.rb
Comment (chrislessard) [reply api/db/migrate/20190516205625_add_columns_to_assignment.rb:5]: I see. Is there any harm in removing them now since they'll be find-able under version control anyways ? I would be tempted to remove them for now and keep a note somewhere when fund_source gets created. We can always put them back.

File: api/app/assets/javascripts/api/v1/assignments.coffee
Comment (chrislessard) [reply api/app/assets/javascripts/api/v1/assignments.coffee]: This looks good to me !

File: api/app/controllers/api/v1/assignments_controller.rb
Comment (chrislessard) [reply api/app/controllers/api/v1/assignments_controller.rb]: Yes !

**Comment by chrislessard** [comment]
@michellemtchai 

Hi Michelle, 

Sorry for the delay. This all looks great to me 🙂Thanks for all your hard work !

**Review by chrislessard — APPROVED** [review]

**Pull request merged**
